### PR TITLE
Show component variants

### PR DESCRIPTION
**Pull request by igloosi**
To aid the usefulness of our accessibility tests we're splitting variants out of the main file into separate files. 
This PR:
* updates app.js to store variants and make them available to the nunjucks template
* updates the component preview and view templates to include the variants section
* updates the gulp generate:readme task to show the same content
* splits button as an example 

@gemmaleigh have a look at the new content structure if it makes sense

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -90,6 +106,17 @@ app.get('/components*', function (req, res) {
   } else {
     // Show the isolated component preview
     path = path.replace(/preview/g, '')
```
Comment (36degrees) [inline app.js]: Why does this happen three times within this else block? Can it not just be done once, e.g. at the end?

File: src/components/button/README.md [hunk 2]
```diff
@@ -2,14 +2,50 @@
...
+
+## Variants
+
+### Button--disabled
+
+[Preview button--disabled variant.](/components/button/button--disabled/preview)
+
+Markup
+
+    <input class="govuk-c-button  " value="Save and continue" >
```
Comment (36degrees) [inline src/components/button/README.md]: This is the same markup as above – it's not disabled…

Also, possibly a bit pedantic, but it'd also be good to try and remove the extra spaces in the classes attribute.

File: src/components/button/README.md [hunk 3]
```diff
@@ -2,14 +2,50 @@
...
+
+## Variants
+
+### Button--disabled
+
+[Preview button--disabled variant.](/components/button/button--disabled/preview)
+
+Markup
```
Comment (36degrees) [inline src/components/button/README.md]: Should these be h4s?

File: app.js [hunk 4]
```diff
@@ -90,6 +106,17 @@ app.get('/components*', function (req, res) {
...
+    if (path.indexOf('--') > -1) {
+      let noLeadingSlash = path.split(/^\/(.+)/)[1]
```
Comment (36degrees) [inline app.js]: This whole block is really hard to follow. Can it be refactored, split up or at least commented?

File: src/components/button/index.njk [hunk 5]
```diff
@@ -7,7 +7,7 @@
 {% block componentDescription %}
   A button is an element that allows users to carry out an action on a GOV.UK page. Common use cases include allowing a user to <b>Start</b> an application or <b>Save and continue</b> their progress.
 
-  A button should have a short text snippet that describes what it will do:
+  A button should have a short text snippet that describes what it will do.
```
Comment (36degrees) [inline src/components/button/index.njk]: What's with the leading spaces here?

File: src/components/details/README.md [hunk 6]
```diff
@@ -10,6 +10,26 @@ Component for conditionally revealing content, using the details HTML element.
...
 
+## Quick start examples
+
+    <details class="govuk-c-details ">
+      <summary class="govuk-c-details__summary">
+        <span class="govuk-c-details__summary-text">Help with nationality</span>
+      </summary>
+      <div class="govuk-c-border govuk-c-border--left-narrow">
+        <div class="govuk-c-details__text">
+          <p>
```
Comment (36degrees) [inline src/components/details/README.md]: It'd be good to get the markup examples indented correctly.

File: tasks/gulp/generate-readme.js [hunk 7]
```diff
@@ -29,6 +29,22 @@ gulp.task('generate:readme', () => {
     vinylInfo.componentName = paths.split(path.sep).slice(-2, -1)[0]
     vinylInfo.componentPath = vinylInfo.componentName
     vinylInfo.componentNunjucksFile = fs.readFileSync(configPath.components + vinylInfo.componentName + '/' + vinylInfo.componentName + '.njk', 'utf8')
+
+    let variantItems = []
+    let files = fs.readdirSync(configPath.components + vinylInfo.componentName + '/')
```
Comment (36degrees) [inline tasks/gulp/generate-readme.js]: What is vinylInfo??

File: src/components/details/README.md (hunk 6 above)
Comment (gemmaleigh) [reply src/components/details/README.md]: I'm going to open a new PR fixing the indentation of component examples.

**Comment by gemmaleigh** [comment]
I'm happy with the structure. 

So we have a component "detail" page here:
https://govuk-frontend-review-pr-235.herokuapp.com/components/button/

The default button component preview
https://govuk-frontend-review-pr-235.herokuapp.com/components/button/preview

and the variants:
https://govuk-frontend-review-pr-235.herokuapp.com/components/button/button--disabled/preview

https://govuk-frontend-review-pr-235.herokuapp.com/components/button/button--start/preview

It doesn't look like the component HTML matches the component example above it, here:
https://govuk-frontend-review-pr-235.herokuapp.com/components/button

I'm not sure that showing the "Quick Start examples" makes sense here, that content should only appear in the README.

Then each component "detail" page should show:

Component default
- Example, HTML, macro, preview link.
Variants
- Variant name
- Example, HTML, macro, preview link.

**Comment by igloosi** [comment]
@36degrees i've refactored the code

**Comment by igloosi** [comment]
also reordered sections
